src/main/java/cn/nukkit/block/Block.java

```diff
@@ -872,7 +883,11 @@ private static double speedRateByHasteLore0(int hasteLoreLevel) {
         return 1.0 + (0.2 * hasteLoreLevel);
     }
 
-    private static int toolType0(Item item) {
+    @PowerNukkitDifference
+    private static int toolType0(Item item, int blockId, int blockToolType) {
+        if((blockId == LEAVES && item.isHoe()) ||
+                (blockId == LEAVES2 && item.isHoe()))
+            return 5;
```

Comment:
Probably better to follow the format and use the ItemTool class for easier readability

Reply:
The ItemTool class doesn't have a simple way to make this work. After several tests, I saw that this was the best way to resolve this issue, considering that this is where the tool verification is done, and not in the ItemTool class.

However, I will do the proper formatting.